docs/Practices/hub.md

```diff
@@ -0,0 +1,80 @@
+- Integration code for 3rd party services should be in a `Data` directory.
+- Repository code (code that interects with a data store), should be in a file called `repository.ts`, OR in a region called `Repository`.
+- (Request or Event) Handlers exposing a command should be in a file called `api.ts`, a region called `API`, or in files called `request.ts` or `event.ts` in a directory called `handlers`.
+- Integration code for bounded contexts in a different project should be in a `dependent-bcs` directory with sub directories matching tha appropriate bounded context. *As of Jan 24 2023 this is not being utilized. Defining this early to reduce decision overhead when this occurs.*
+
+Tests:
+- Test files should sit close to the code they test. *Test files that test code in disparate files should be broken up into multiple test files itself.*
+
+SQL:
+- Avoid using the (massive scripts directory)[https://massivejs.org/docs/functions-and-scripts]. Prefer instead to define a query as close to it's consumers as possible. *Defining queries in the scripts directory opens up the possibility of unintended coupling by non cohesive modules. There are solutions to syntax highlighting that don't require a .sql file.*
+
+## The Future
+- If and as we collect more data/it grows in complexity - we likely will need to apply the same decoupling of domains by way of bounded contexts to our databases. *We should condsider breaking up the database. Technical requirements and prudence should drive the implementation of breakup (schemas vs instances, etc). Each new distinct piece should relate to a given bouned context and should house ONLY the data required to support the Vertical Slices of those BCs.*
+
+## Additional Principles
+These principles, while value adding, do not directly support the goals nor did they fall into non-goal. As such guidelines were not built around these principles. These 
+
+- Validate at all IO boundaries (guideline above indicates just handler request data). *Typescript can be a powerful tool, but it is only useful at compile time (as you write and build code). Validating that data being ingested by a process written using Typescript can go a long way in ensuring that the type safety provided by Typescript is reliable.*
+
+- Endeavour to make invalid states impossible. *Example1: If performing operations that utilize an email address which must be validated, consider creating a branded type `type ValidatedEmail = { _tag: "ValidatedEmail"; value: string }` rather than utilizing a `string`. Example2: Rather than `type School = { name: string; familyData?: FamilyData; partnerData?: PartnerData }` do `type School = FamilySchool | PartnerSchool; type FamilySchool = { _tag: "FamilySchool"; name: string; familyData: FamilyData }; type PartnerSchool = { _tag: "PartnerSchool"; name: string; partnerData: PartnerData }`*
```

Comment:
Can you move all these examples into their own ``` blocks so it's broken up a bit better on render?

Reply:
examples have their own file now